tiled/client/base.py

```diff
@@ -121,6 +121,39 @@ def formats(self):
         )
         return sorted(formats)
 
+    def update_metadata(self, metadata=None, specs=None):
+        """
+        EXPERIMENTAL: Update metadata.
+
+        This is subject to change or removal without notice
+
+        Parameters
+        ----------
+        metadata : dict, optional
+            User metadata. May be nested. Must contain only basic types
+            (e.g. numbers, strings, lists, dicts) that are JSON-serializable.
+        specs : List[str], optional
+            List of names that are used to label that the data and/or metadata
+            conform to some named standard specification.
+        """
+
+        self._cached_len = None
+
+        data = {  # noqa: F841
```

Comment:
This F841 seems off. The variable is used below.

My guess is that you were trying to commit partial work, when you planned to use the variable but were not yet using it. In that situation, it is useful to know about 

```
git commit --no-verify ...
```

which disables the pre-commit hooks and enables you to commit partial work. This is IMO better than putting # noqa exemptions in place temporarily because it is easy to forget to remove them, as may have happened in this case.